packages/dds/tree/src/test/shared-tree/undo.spec.ts

```diff
@@ -453,6 +464,137 @@ describe("Undo and redo", () => {
+			redoStack: redoStack2,
+			unsubscribe: unsubscribe2,
+		} = createTestUndoRedoStacks(forkedView.events);
+
+		assert(forkedView.root.child !== undefined);
+		forkedView.root.child.propertyOne = 512; // 256 -> 512
+
+		undoStack2.pop()?.revert();
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+
+		const undoOriginalPropertyTwo = undoStack1.pop();
+		const clonedUndoOriginalPropertyTwo = undoOriginalPropertyTwo?.clone(forkedView);
+
+		const undoOriginalPropertyOne = undoStack1.pop();
+		const clonedUndoOriginalPropertyOne = undoOriginalPropertyOne?.clone(forkedView);
+
+		undoOriginalPropertyOne?.revert();
+		undoOriginalPropertyTwo?.revert();
+
+		assert.equal(originalView.root.child?.propertyOne, 128);
+		assert.equal(originalView.root.child?.propertyTwo.itemOne, "");
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "newItem");
+
+		clonedUndoOriginalPropertyOne?.revert();
+		clonedUndoOriginalPropertyTwo?.revert();
+
+		assert.equal(forkedView.root.child?.propertyOne, 128);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "");
+
+		assert.equal(undoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(undoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+	});
+
+	it("reverts cloned revertible on original view", () => {
+		const originalView = asTreeViewAlpha(createLocalSharedTree("testSharedTree"));
+
+		const {
+			undoStack: undoStack1,
+			redoStack: redoStack1,
+			unsubscribe: unsubscribe1,
+		} = createTestUndoRedoStacks(originalView.events);
+
+		assert(originalView.root.child !== undefined);
+		originalView.root.child.propertyOne = 256; // 128 -> 256
+		originalView.root.child.propertyTwo.itemOne = "newItem";
+
+		const undoOriginalPropertyOne = undoStack1.pop();
+		const undoOriginalPropertyTwo = undoStack1.pop();
+
+		const clonedUndoOriginalPropertyOne = undoOriginalPropertyOne?.clone(originalView);
+		const clonedUndoOriginalPropertyTwo = undoOriginalPropertyTwo?.clone(originalView);
+
+		clonedUndoOriginalPropertyOne?.revert();
+		clonedUndoOriginalPropertyTwo?.revert();
+
+		assert.equal(originalView.root.child?.propertyOne, 128);
+		assert.equal(originalView.root.child?.propertyTwo.itemOne, "");
+		assert.equal(undoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(undoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+	});
+
+	it("clone revertible fails if trees are different", () => {
+		const viewA = asTreeViewAlpha(createLocalSharedTree("testSharedTreeOne"));
+		const viewB = asTreeViewAlpha(createLocalSharedTreeTwo("testSharedTreeTwo"));
```

Comment:
For this test case, it's not important that the two trees have different _schema_ - the part that matters is that they are different instances of a SharedTree. So, it should be perfectly fine to initialize `viewB` using a second call to `createLocalSharedTree()` rather than `createLocalSharedTreeTwo()`. And then you can delete `createLocalSharedTreeTwo()`.